pyanaconda/network.py

```diff
@@ -272,6 +272,55 @@ def unset(self, *args):
         ifcfglog.debug("IfcfgFile.unset %s: %s", self.filename, args)
         SimpleConfigFile.unset(self, *args)
 
+def ensure_single_initramfs_connections():
+    """Ensure device configured in initramfs has no more than one NM connection.
+
+    In case of multiple connections for device having ifcfg configuration from
+    boot options, the connection should correspond to the ifcfg file.
+    NetworkManager can be generating additional in-memory connection in case it
+    fails to match device configuration to the ifcfg (#1433891).  By
+    reactivating the device with ifcfg connection the generated in-memory
+    connection will be deleted by NM.
+
+    Don't enforce on slave devices for which having multiple connections can be
+    valid (slave connection, regular device connection).
+    """
+
+    rv = []
+
+    for dev_name in nm.nm_devices():
+        try:
+            nm.nm_device_setting_value(dev_name, "connection", "uuid")
+        except nm.SettingsNotFoundError:
+            pass
+        except nm.MultipleSettingsFoundError:
+            if nm.nm_device_is_slave(dev_name):
+                continue
+            ifcfg_path = find_ifcfg_file_of_device(dev_name)
+            if ifcfg_path:
```

Comment:
I think there could be 'device_has_ifcfg_from_kickstart()' for this first ``if``, ``with`` and the second ``if``. And then the part below ``if not "Generated by parse-kickstart" in f.read()`` could go into a function called ``ensure_device_activated()`` or something like that.